src/ConnectionAutomaton.cpp

```diff
@@ -109,10 +121,14 @@ void ConnectionAutomaton::closeDuplexConnection(folly::exception_wrapper ex) {
     connectionOutput_.onComplete();
   }
   connectionInputSub_.cancel();
-  connection_.reset();
+}
 
-  stats_.socketClosed();
-  onClosed_();
+void ConnectionAutomaton::closeWithError(Frame_ERROR&& error) {
+  VLOG(4) << "closeWithError "
+          << error.payload_.data->cloneAsValue().moveToFbString();
+
+  outputFrameOrEnqueue(error.serializeOut());
+  close(folly::exception_wrapper(), StreamCompletionSignal::ERROR);
```

Comment:
folly::exception_wrapper() seems like it can be interpreted multiple ways

https://github.com/facebook/folly/blob/master/folly/ExceptionWrapper.h#L72

```
 * // Thread1
 * void doSomethingCrazy() {
 *   int rc = doSomethingCrazyWithLameReturnCodes();
 *   if (rc == NAILED_IT) {
 *     globalExceptionWrapper = exception_wrapper();
 *   } else if (rc == FACE_PLANT) {
 *     globalExceptionWrapper = make_exception_wrapper<FacePlantException>();
```

https://github.com/facebook/folly/blob/master/folly/ExceptionWrapper.h#L155